src/utils/node_byte_utils.ts

```diff
@@ -125,8 +125,8 @@ export const nodeJsByteUtils = {
     return Buffer.from(text, 'utf8');
   },
 
-  toUTF8(buffer: Uint8Array): string {
-    return nodeJsByteUtils.toLocalBufferType(buffer).toString('utf8');
+  toUTF8(buffer: Uint8Array, start?: number, end?: number): string {
+    return (buffer as NodeJsBuffer).toString('utf8', start, end);
```

Comment:
I don't like the cast here, but I'm not sure how to go about changing this in a way that wouldn't require a large refactoring of the entire `ByteUtils` interface.

Running the tests, it's apparent that wherever `ByteUtils.toUTF8` is called that the buffer passed in is already a `NodeJsBuffer`, so it begs the question: where else we should be able to get rid of calls to `toLocalBufferType` and how much more speedup we can get as a result?

Reply:
Yeah, I don’t like the cast either, it’s making me a bit that this would just silently give the wrong result if `buffer` is a plain `Uint8Array` (like the method signature asks for!).

> that wouldn't require a large refactoring of the entire `ByteUtils` interface

Honestly, I think this might be the way to go, even if it’s inconvenient.

```js
export type ByteUtils<LBT extends Uint8Array> = {
  toLocalBufferType(buffer: Uint8Array | ArrayBufferView | ArrayBuffer): LBT;
  fromUTF8: (text: string) => LBT;
  toUTF8: (buffer: LBT) => string;
  /* ... */
};

export const webByteUtils: ByteUtils<Uint8Array> = {
  toLocalBufferType(buffer: Uint8Array | ArrayBufferView | ArrayBuffer): Uint8Array;
  fromUTF8: (text: string) => Uint8Array;
  toUTF8: (buffer: Uint8Array) => string;
  /* ... */
};

export const nodeJsByteUtils: ByteUtils<Buffer> = {
  toLocalBufferType(buffer: Uint8Array | ArrayBufferView | ArrayBuffer): Buffer;
  fromUTF8: (text: string) => Buffer;
  toUTF8: (buffer: Buffer) => string;
  /* ... */
};
```

Reply:
I'm also concerned about the cast here and think it's probably better to go with something like what Anna has recommended here just to ensure we've got proper typing in place. It's an internal class so we should feel free to update as needed.

Reply:
I discussed this with @nbbeeken and we came to the conclusion that a refactor on the scale that would be necessary to get this to work would be out of scope for this ticket.

I think it would be better to leave the call to `toLocalBufferType` in `toUTF8` and address that in a later PR. For now, the change that this PR will centre around is the removal of unnecessary `Buffer.subarray` calls at `ByteUtils.toUTF8` callsites. I will update the PR description and ticket description and post updated performance results with the new changes.